hub/core/storage/s3/s3_mapper.py

```diff
@@ -0,0 +1,173 @@
+            config=self.client_config,
+            endpoint_url=self.endpoint_url,
+            region_name=self.aws_region,
+        )
+
+    def __setitem__(self, path, content):
+        """
+        Sets the object present at the path with the value
+
+        Args:
+            path (str): the path relative to the root of the mapper.
+            content (bytes): the value to be assigned at the path.
+
+        Returns:
+            None
+
+        Raises:
+            S3SetError: Any S3 error encountered while setting the value at the path.
+        """
+        try:
+            path = posixpath.join(self.path, path)
+            content = bytearray(memoryview(content))
+            attrs = {
+                "Bucket": self.bucket,
+                "Body": content,
+                "Key": path,
+                "ContentType": ("application/octet-stream"),
+            }
+            self.client.put_object(**attrs)
+        except Exception as err:
+            raise S3SetError(err)
+
+    def __getitem__(self, path):
+        """
+        Gets the object present at the path.
+
+        Args:
+            path (str): the path relative to the root of the mapper.
+
+        Returns:
+            bytes: The bytes of the object present at the path.
+
+        Raises:
+            KeyError: If an object is not found at the path.
+            S3GetError: Any other error other than KeyError while retrieving the object.
+        """
+        try:
+            path = posixpath.join(self.path, path)
+            resp = self.client.get_object(
+                Bucket=self.bucket,
+                Key=path,
+            )
+            return resp["Body"].read()
+        except ClientError as err:
+            if err.response["Error"]["Code"] == "NoSuchKey":
+                raise KeyError(err)
+            raise
+        except Exception as err:
+            raise S3GetError(err)
+
+    def __delitem__(self, path):
+        """
+        Delete the object present at the path.
+
+        Args:
+            path (str): the path to the object relative to the root of the mapper.
+
+        Returns:
+            None
+
+        Raises:
+            S3DeletionError: Any S3 error encountered while deleting the object. Note: if the object is not found, s3 won't raise KeyError.
+        """
+        try:
+            path = posixpath.join(self.path, path)
+            self.client.delete_object(Bucket=self.bucket, Key=path)
+        except Exception as err:
+            raise S3DeletionError(err)
+
+    def _list_objects(self):
+        """
+        Helper function to list all the objects present at the root of the mapper.
+
+        Args:
+            None
+
+        Returns:
+            list: list of all the objects found at the root of the mapper.
+
+        Raises:
+            S3ListError: Any S3 error encountered while listing the objects.
+        """
+        try:
+            # TODO boto3 list_objects only returns first 1000 objects
```

Comment:
Here's a nice blog post on how to resolve this issue. No additional libraries needed!

https://adamj.eu/tech/2018/01/09/using-boto3-think-pagination/